pkg/assets/file_builder.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assets
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"k8s.io/kops/pkg/apis/kops"
+)
+
+// AssetBuilder discovers and remaps assets
+type FileAssetBuilder struct {
+	FileAssets     []*FileAsset
+	AssetsLocation *kops.Assets
+}
+
+type FileAsset struct {
+	// File will be the name of the file we should use
+	File string
+
+	// CanonicalLocation will be the source location of the file, if we should copy it to the actual location
+	CanonicalLocation string
+}
+
+func NewFileAssetBuilder(assets *kops.Assets) *FileAssetBuilder {
+	return &FileAssetBuilder{
+		AssetsLocation: assets,
+	}
+}
+
+// TODO SHA
+// TODO need to map file in the same manner and include vfs and http location
```

Comment:
Open question I have, and I updated the comment, is do we have the SHA url with the file URL.